netmiko/nokia/nokia_sros_ssh.py

```diff
@@ -168,3 +170,95 @@ def cleanup(self, command="logout"):
         # Always try to send final 'logout'.
         self._session_log_fin = True
         self.write_channel(command + self.RETURN)
+
+
+class NokiaSrosFileTransfer(BaseFileTransfer):
+    def _get_cmd_prefix(self):
+        """
+        Returns "//" if the current prompt is MD-CLI
+        empty string otherwise
+        """
+        return "//" if "@" in self.ssh_ctl_chan.base_prompt else ""
+
+    def remote_space_available(self, search_pattern=r"(\d+)\s+\w+\s+free"):
+        """Return space available on remote device."""
+
+        # Sample text for search_pattern.
+        # "               3 Dir(s)               961531904 bytes free."
+        self.ssh_ctl_chan.send_command(self._get_cmd_prefix() + "environment no more")
+        remote_cmd = self._get_cmd_prefix() + "file dir {}".format(self.file_system)
+        remote_output = self.ssh_ctl_chan.send_command(remote_cmd)
+        match = re.search(search_pattern, remote_output)
+        return int(match.group(1))
+
+    def check_file_exists(self, remote_cmd=""):
+        """Check if destination file exists (returns boolean)."""
+
+        if self.direction == "put":
+            self.ssh_ctl_chan.send_command(
+                self._get_cmd_prefix() + "environment no more"
+            )
```

Comment:
Should be unneeded.

Reply:
correct, I missed this.
fixed the logic by disabling the paging for both CLI engines in the session preparation (done in bfcd0a8)

With this change there is no more need to disable the paging in the FileTransfer class.